app/celery/process_delivery_status_result_tasks.py

```diff
@@ -34,222 +33,57 @@
 )
 @statsd(namespace='tasks')
 def process_delivery_status(
-    self,
+    self: Task,
     event: CeleryEvent,
-) -> bool:
+) -> None:
     """
-    This is a Celery task for updating the delivery status of a notification.
+    This is a Celery task for updating the delivery status of a Twilio notification.
     """
 
-    # preset variables to address "unbounded local variable"
-    sqs_message = None
-    notification_platform_status = None
-
-    current_app.logger.info('processing delivery status')
-    current_app.logger.debug(event)
+    current_app.logger.debug('twilio incoming sms update: %s', event)
 
-    # first attempt to process the incoming event
-    sqs_message = _get_sqs_message(event)
-
-    # get the provider
-    (provider_name, provider) = _get_provider_info(sqs_message)
+    try:
+        sqs_message = _get_sqs_message(event)
+        current_app.logger.debug('twilio decoded sms update: %s', sqs_message)
+        provider_name, provider = _get_provider_info(sqs_message)
+    except NonRetryableException:
+        statsd_client.incr('clients.sms.twilio.status_update.error')
+        raise
 
-    body = sqs_message.get('body')
-    current_app.logger.info('retrieved delivery status body: %s', body)
+    notification_platform_status: SmsStatusRecord = get_notification_platform_status(provider, sqs_message.get('body'))
```

Comment:
NIT: (and likely not apart of this PR) The code path for `get_notification_platform_status` is dead and will result in a `NotImplementedError` error.

Reply:
@k-macmillan I'd like to talk about how the `notification_platform_status` variable is being used further down the code path. Is this being populated as expected?